src/libsync/creds/oauth.cpp

```diff
@@ -249,32 +261,43 @@ void OAuth::startAuthentication()
 
     connect(this, &OAuth::fetchWellKnownFinished, this, [this] {
         if (Theme::instance()->oidcEnableDynamicRegistration() && _registrationEndpoint.isValid()) {
-            auto job = new RegisterClientJob(_networkAccessManager, _dynamicRegistrationData, _registrationEndpoint, this);
-            connect(job, &RegisterClientJob::finished, this, [this](const QString &clientId, const QString &clientSecret, const QVariantMap &dynamicRegistrationData) {
+            auto registerJob = new RegisterClientJob(_networkAccessManager, _dynamicRegistrationData, _registrationEndpoint, this);
+            connect(registerJob, &RegisterClientJob::finished, this, [this](const QString &clientId, const QString &clientSecret, const QVariantMap &dynamicRegistrationData) {
+                qCDebug(lcOauth) << "client registration finished successfully";
                 _clientId = clientId;
                 _clientSecret = clientSecret;
                 Q_EMIT dynamicRegistrationDataReceived(dynamicRegistrationData);
                 Q_EMIT authorisationLinkChanged();
             });
-            connect(job, &RegisterClientJob::errorOccured, this, [this](const QString &error) {
+            connect(registerJob, &RegisterClientJob::errorOccured, this, [this](const QString &error) {
                 qCWarning(lcOauth) << "Failed to dynamically register the client, try the default client id" << error;
                 Q_EMIT authorisationLinkChanged();
             });
-            job->start();
+            registerJob->start();
         } else {
+            qCDebug(lcOauth) << "dynamic registration disabled and/or registration endpoint invalid";
```

Comment:
hmm better log depending on the case.
Also its not invalid but not present.

Reply:
This is a direct reference to the `if`'s `isValid` call. I guess I'll add a docstring to clarify the intention of this call.